### PR TITLE
ddl: fix index recognized as prefix index when column length is enlarged

### LINKED ISSUE #5134: Incorrect prefix index marking
```sql
create table city (
    city VARCHAR(2) KEY
);

alter table city change column city city varchar(50);

insert into city values ("abc"), ("abd");
```

Error:

ERROR 1062 (23000): Duplicate entry 'abd' for key 'PRIMARY'

This is because that when using the inline KEY syntax, we set its length to 2 instead of -1.

### PR DESCRIPTION
**Pull request by breeswish**
fix #5134 

@zimulala @winoros PTAL

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -263,10 +263,6 @@ func columnDefToCol(ctx context.Context, offset int, colDef *ast.ColumnDef) (*ta
 	if colDef.Options != nil {
 		len := types.UnspecifiedLength
```
Comment (shenli) [inline ddl/ddl_api.go]: Should we consider prefix-index here?

Comment (mikumikuchan) [reply ddl/ddl_api.go]: According to the syntax, it seems to be impossible to specify a prefix-index here:

```
column_definition:
    data_type [NOT NULL | NULL] [DEFAULT default_value]
      [AUTO_INCREMENT] [UNIQUE [KEY]] [[PRIMARY] KEY]
      [COMMENT 'string']
      [COLUMN_FORMAT {FIXED|DYNAMIC|DEFAULT}]
      [STORAGE {DISK|MEMORY|DEFAULT}]
      [reference_definition]
  | data_type [GENERATED ALWAYS] AS (expression)
      [VIRTUAL | STORED] [NOT NULL | NULL]
      [UNIQUE [KEY]] [[PRIMARY] KEY]
      [COMMENT 'string']
```

https://dev.mysql.com/doc/refman/5.7/en/create-table.html

Comment (breeswish) [reply ddl/ddl_api.go]: According to the syntax, it should be impossible to specify a prefix-index in CREATE TABLE column definition:

```
column_definition:
    data_type [NOT NULL | NULL] [DEFAULT default_value]
      [AUTO_INCREMENT] [UNIQUE [KEY]] [[PRIMARY] KEY]
      [COMMENT 'string']
      [COLUMN_FORMAT {FIXED|DYNAMIC|DEFAULT}]
      [STORAGE {DISK|MEMORY|DEFAULT}]
      [reference_definition]
  | data_type [GENERATED ALWAYS] AS (expression)
      [VIRTUAL | STORED] [NOT NULL | NULL]
      [UNIQUE [KEY]] [[PRIMARY] KEY]
      [COMMENT 'string']
```

https://dev.mysql.com/doc/refman/5.7/en/create-table.html

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
please cherry-pick to 1.0

**Pull request merged**